### PR TITLE
SPP12087 help centre structure refactor

### PR DESCRIPTION
**Pull request by JasonBellONS**
Help centre refactored to no longer pull in the structure page from contentful, structure and navigation built on demand using information from the other help centre pages grabbed from contentful

### CODE REVIEW

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by JasonBellONS** [comment]
Note failing tests due to changes in ordering of the content, should be resolved under https://jira.ons.gov.uk/browse/SPP-11740

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Comment by JasonBellONS** [comment]
> Not reviewed the code yet, but could we try and maintain URLs. When I load help centre pages in the dev server they have a different URL to what we currently had deployed. E.g:
> 
> **Find and view methods** prod : image
> 
> dev : image
> 
> If you could check the pages and try and get them to align that'd be good.

Hi, this should now be fixed

**Review by gibbardsteve — COMMENTED** [review]
A few questions and some minor comments. 

Also I notice the live site doesn't have Help Centre Index in Alphabetical order:

image

I vaguely recall this may have been a bug we noticed in the last deploy that was since fixed (i.e made alphabetical) - just checking the change is expected?

**Comment by JasonBellONS** [comment]
> A few questions and some minor comments.
> 
> Also I notice the live site doesn't have Help Centre Index in Alphabetical order:
> 
> image
> 
> I vaguely recall this may have been a bug we noticed in the last deploy that was since fixed (i.e made alphabetical) - just checking the change is expected?

If i remember correctly it was the methods page that was made alphabetical, i made a similar change here as you can't ensure that content is gathered in a specific order, my thoughts were displaying it alphabetically made most sense

File: sml_builder/__init__.py
Comment (JasonBellONS) [reply sml_builder/__init__.py:129]: cache is a dictionary object within the flask application object, each time we deploy that application will be recreated, so in the case of this method we deploy, and during the first request to it we create this navigation and add it to the cache dictionary, it will remain here unless deleted or the app is stopped

File: sml_builder/__init__.py
Comment (JasonBellONS) [reply sml_builder/__init__.py:90]: any request to open a page on the website, so a user going to the home page or any subsequent page would trigger it, after the first run though the method is removed from that declaration

**Comment by JasonBellONS** [comment]
@gibbardsteve comments added where requested

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Review by gibbardsteve — APPROVED** [review]
Approved. We may want to hold off merging until we get a green light from MQD.

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d1uzhgtw86gcqp.cloudfront.net/
cloudfront_id=EEJOC5TWG5769

**Pull request merged**
